### PR TITLE
Add Renewer Cronjob

### PR DESCRIPTION
**Pull request by paurosello**
This PR:

### Adds
- Cronjob to renew karpenter nodes when other workers in the cluster are upgrading

### CODE REVIEW

**Comment by paurosello** [comment]
Example run

```
Starting node renewal process...
Fetching the highest version of nodes without the 'managed-by: karpenter' label...
Target version identified as: 14.22.0
11 total nodes managed by Karpenter.
Fetching nodes with the 'managed-by: karpenter' label and a different version...
5 nodes identified with a different version.
ip-10-35-10-205.us-west-2.compute.internal ip-10-35-10-51.us-west-2.compute.internal ip-10-35-13-56.us-west-2.compute.internal ip-10-35-15-104.us-west-2.compute.internal ip-10-35-15-181.us-west-2.compute.internal
Planning to delete 2 nodes.
Deleting node: ip-10-35-10-205.us-west-2.compute.internal
node "ip-10-35-10-205.us-west-2.compute.internal" deleted
Deleting node: ip-10-35-10-51.us-west-2.compute.internal
node "ip-10-35-10-51.us-west-2.compute.internal" deleted
Node renewal process completed.
Stream closed EOF for kube-system/karpenter-renewer-manual-4px-s6lcq (renewer)
```

File: helm/karpenter/templates/node-renewer.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+              echo "Planning to delete $NUM_NODES_TO_DELETE nodes."
+
+              # Get the list of nodes to delete
+              NODES_TO_DELETE=$(echo "$NODES_TO_DELETE" | head -n $NUM_NODES_TO_DELETE)
+
+              # Delete the nodes
+              for NODE in $NODES_TO_DELETE; do
+                  echo "Deleting node: $NODE"
+                  kubectl delete node $NODE --wait=false
```
Comment (whites11) [inline helm/karpenter/templates/node-renewer.yaml:128]: this is enough for karpenter to get rid of the VM?

**Review by whites11 — APPROVED** [review]
LGTMIIWFY

File: helm/karpenter/templates/node-renewer.yaml (hunk 1 above)
Comment (paurosello) [reply helm/karpenter/templates/node-renewer.yaml:128]: yeh, karpenter will drain and delete node

**Pull request merged**